bigchaindb/backend/rethinkdb/config.py

```diff
@@ -0,0 +1,48 @@
+"""Database configuration functions."""
+import logging
+from functools import partial
+
+import rethinkdb as r
+
+from bigchaindb.backend import config
+from bigchaindb.backend.exceptions import DatabaseOpFailedError
+from bigchaindb.backend.rethinkdb.connection import RethinkDBConnection
+
+logger = logging.getLogger(__name__)
+
+TABLES = ('bigchain', 'backlog', 'votes')
+
+
+def _configure_table_helper(connection, table, *, replicas=None, shards=None):
+    try:
+        connection.run(
+            r.table(table).reconfigure(shards=shards, replicas=replicas)
+        )
+    except r.ReqlOpFailedError as e:
+        raise DatabaseOpFailedError from e
+
+
+def _configure_table(connection, table, *, replicas=None, shards=None):
+    if replicas is None:
+        table_config = connection.run(r.table(table).config())
+        replicas = len(table_config['shards'][0]['replicas'])
+    if shards is None:
+        table_config = connection.run(r.table(table).config())
+        shards = len(table_config['shards'])
+    _configure_table_helper(
+        connection, table, replicas=replicas, shards=shards)
+
+_configure_table_shards = partial(_configure_table, replicas=None)
+_configure_table_replicas = partial(_configure_table, shards=None)
```

Comment:
> The code for each, set_shards and set_replicas is nearly identical, and very repetitive. So this was an attempt to limit the duplication of identical logic.

Ok, I see where you're going with this, but now the logic's being duplicated across two if statements in `_configure_table()`. Maybe this was more because of a lack of time?

For better or for worse, here's how I would reformulate this:

```python
def get_table_config(connection, table):
    return connection.run(r.table(table).config())


def reconfigure_table(connection, table, **kwargs):
    try:
        connection.run(r.table(table).reconfigure(**kwargs)
    except r.ReqlOpFailedError as e:
        raise DatabaseOpFailedError from e


@config.set_shards.register(RethinkDBConnection)
def set_shards(connection, shards):
    for table in TABLES:
        config = get_table_config(connection, table)
        replicas = len(table_config['shards'][0]['replicas'])
        reconfigure_table(connection, table, replicas=replicas, shards=shards)

@config.set_replicas.register(RethinkDBConnection)
def set_replicas(connection, replicas):
    for table in TABLES:
        config = get_table_config(connection, table)
        shards = len(table_config['shards'])
        reconfigure_table(connection, table, replicas=replicas, shards=shards)
```

It's kind of a pain you have to be explicit about the other setting (e.g. you have to set `replicas` to be the same as before if you're only changing the `shards`); is this necessary @r-marques?